memory/src/main/scala/filodb.memory/format/vectors/DeltaDeltaVector.scala

```diff
@@ -274,12 +270,21 @@ object DeltaDeltaConstDataReader extends LongVectorDataReader {
     }
   }
 
-  def changes(vector: BinaryVectorPtr, start: Int, end: Int): Int = {
+  def changes(vector: BinaryVectorPtr, start: Int, end: Int, prev: Long, ignorePrev: Boolean = false): (Long, Long) = {
+
+    require(start >= 0 && end < length(vector), s"($start, $end) is out of bounds, length=${length(vector)}")
+    val firstValue = iterate(vector, start).next
+    val lastValue = iterate(vector, end).next
```

Comment:
You can call `apply` instead of `iterate`, this avoids the creation of two objects.  Also you don't need to call lastValue all the time.   So something like this:

    val firstValue = apply(vector, start)
    val changes = if (!ignorePrev && prev != firstValue) 1 else 0
    if (slope(vector) == 0) (0 + changes, apply(vector, end))